test/watch.js

```diff
@@ -696,22 +708,110 @@ describe('Query watch', function() {
         })
         .then(() => {
           return streamHelper.awaitOpen();
+        })
+        .then(() => {
+          streamHelper.close();
+          return streamHelper.await('end');
+        })
+        .then(() => {
+          return streamHelper.awaitOpen();
+        })
+        .then(() => {
+          assert.equal(streamHelper.streamCount, 3);
         });
     });
   });
 
-  it("doesn't re-open without resume token", function() {
-    watchHelper.startWatch();
+  it("doesn't re-open inactive stream", function() {
+    // This test uses the normal timeout handler since we unsubscribe from the
+    // Watch stream while the stream is recovering from an error.
```

Comment:
I don't really understand this comment.  Why does unsubscribing while the watch stream is recovering mean we have to use setTimeout instead of setImmediate?

Reply:
I improved the comment, but this test is my pride since it executes the following line:

```
      self._backoff.backoffAndWait().then(() => {
        if (!isActive) {
          Firestore.log('Watch.onSnapshot', 'Not initializing inactive stream');
          return;
        }
```